src/wallet/wallet2.cpp

```diff
@@ -4918,11 +5048,20 @@ std::vector<std::pair<crypto::key_image, crypto::signature>> wallet2::export_key
 
     crypto::public_key tx_pub_key = get_tx_pub_key_from_received_outs(td);
 
+    // onetime address scheme
+    payment_id_info pID_info;
+    get_payment_id_info(tx_extra_fields, m_account.get_keys().m_view_secret_key, pID_info);
+
     // generate ephemeral secret key
-    crypto::key_image ki;
-    cryptonote::keypair in_ephemeral;
-    cryptonote::generate_key_image_helper(m_account.get_keys(), tx_pub_key, td.m_internal_output_index, in_ephemeral, ki);
+    crypto::key_image ki, ki2;
+    cryptonote::keypair in_ephemeral, in_ephemeral2;
+    cryptonote::generate_key_image_helper_onetime(m_account.get_keys(), tx_pub_key, pID_info.onetime_c, td.m_internal_output_index, in_ephemeral, ki, in_ephemeral2, ki2);
 
+    if (td.m_key_image_known && ki != td.m_key_image || in_ephemeral.pub != pkey)
+    {
+      in_ephemeral = in_ephemeral2;
+      ki = ki2;
+    }
```

Comment:
This seems wrong. I think It should be:
    if (in_ephemeral.pub != pkey)
    {
      in_ephemeral = in_ephemeral2;
      ki = ki2;
    }